include/model.php

```diff
@@ -231,10 +168,9 @@ public function set_languages_ready() {
 	 * @param string $taxonomy taxonomy name
 	 * @return void
 	 */
-	public function clean_languages_cache( $term = 0, $taxonomy = null ) {
+	public function clean_languages_cache( $term = 0, $taxonomy = null ): void {
 		if ( empty( $taxonomy ) || 'language' === $taxonomy ) {
-			delete_transient( 'pll_languages_list' );
-			$this->cache->clean();
+			$this->languages_model->clean_cache();
```

Comment:
Any reason to keep this method here and not to move it *completely* to the Languages class?

Reply:
Yes: the one from the languages class has no parameters, whose are used only for the hooks. It's a sort of decoupling the languages class from the WP hooks.